### PR TITLE
Playlists Page

### PR DESCRIPTION
**Pull request by foolhardy21**
## Playlists Page.

This PR has playlists page APIs and UI components. User has to be logged in to use the feature. 

- Files to review:
  - [ ] src/components/Playlists/PlaylistVideoCard.js
  - [ ] src/components/Playlists/PlaylistsSection.js
  - [ ] src/components/Videos/PlaylistModal.js
  - [ ] src/components/Videos/VideoCard.js
  - [ ] src/contexts/playlists.context.js
  - [ ] src/reducers/playlists.reducer.js

- Use this link for preview - https://deploy-preview-6--pensive-hodgkin-f2cf81.netlify.app/playlists

v-1
v-2
v-3
Thanks

### CODE REVIEW

File: src/components/Videos/PlaylistModal.js [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+
+const PlaylistModal = () => {
+    const [selectedPlaylist, setSelectedPlaylist] = useState({})
+    const { theme } = useTheme()
+    const { playlistsState: { playlists }, playlistsDispatch, getPlaylists, addVideoToPlaylist, hidePlaylistModal } = usePlaylists()
+    const { selectedVideo, showVideosAlert } = useVideos()
+
+    useEffect(() => {
+        (async () => {
+            const getPlaylistsResponse = await getPlaylists()
```
Comment (vishal804) [inline src/components/Videos/PlaylistModal.js:14]: Suggestion: Can Use try catch as it will help in handle error in fetch data

**Review by vishal804 — APPROVED** [review]
Code is readable and clean Nice Work

File: src/pages/Playlists.js [hunk 2]
```diff
@@ -20,7 +20,10 @@ const Playlists = () => {
             }
         })()
     }, [])
-    console.log(playlistsState.loading)
+
+
+    console.log('renderd')
```
Comment (MayankKumar10) [inline src/pages/Playlists.js]: Suggestion: Remove Consol log from here

**Review by MayankKumar10 — APPROVED** [review]
Look into suggestion else everything looks good to me you can merge it.

File: src/components/Videos/PlaylistModal.js (hunk 1 above)
Comment (foolhardy21) [reply src/components/Videos/PlaylistModal.js:14]: yes I have used try catch in `getPlaylists()`

File: src/pages/Playlists.js (hunk 2 above)
Comment (foolhardy21) [reply src/pages/Playlists.js]: removed. thanks.

**Pull request merged**